src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1195,17 +1465,739 @@ private static void ReportReadOnlyError(Symbol symbol, SyntaxNode node, BindValu
+
+                    var escape = GetInvocationEscapeScope(
+                        null,
+                        constructorSymbol.Parameters,
+                        objectCreation.Arguments,
+                        objectCreation.ArgumentRefKindsOpt,
+                        objectCreation.ArgsToParamsOpt,
+                        scopeOfTheContainingExpression);
+
+                    //PROTOTYPE(span): handle object initializers. (no neeed to worry about collection init, the obj must be IEnumerable)
+                    return escape;
+
+                //PROTOTYPE(span): add tests for the following cases
+
+                case BoundKind.UnaryOperator:
+                    return GetValEscape(((BoundUnaryOperator)expr).Operand, scopeOfTheContainingExpression);
+
+                case BoundKind.Conversion:
+                    var conversion = (BoundConversion)expr;
+
+                    //PROTOTYPE(span): currently span-typed stackalloc is wrapped into a conversion node.
+                    //                 this check may not be needed if that changes
+                    if (conversion.ConversionKind == ConversionKind.StackAllocToSpanType)
+                    {
+                        return Binder.TopLevelScope;
+                    }
+
+                    return GetValEscape(((BoundConversion)expr).Operand, scopeOfTheContainingExpression);
+
+                case BoundKind.AssignmentOperator:
+                    return GetValEscape(((BoundAssignmentOperator)expr).Right, scopeOfTheContainingExpression);
+
+                case BoundKind.IncrementOperator:
+                    return GetValEscape(((BoundIncrementOperator)expr).Operand, scopeOfTheContainingExpression);
+
+                case BoundKind.CompoundAssignmentOperator:
+                    var compound = (BoundCompoundAssignmentOperator)expr;
+
+                    return Math.Max(GetValEscape(compound.Left, scopeOfTheContainingExpression),
+                                    GetValEscape(compound.Right, scopeOfTheContainingExpression));
+
+                case BoundKind.BinaryOperator:
+                    var binary = (BoundBinaryOperator)expr;
+
+                    return Math.Max(GetValEscape(binary.Left, scopeOfTheContainingExpression),
+                                    GetValEscape(binary.Right, scopeOfTheContainingExpression));
+
+                    //PROTOTYPE(span): handle remaining expressions with operands - ...
+                    //      Note that requirement that the result is ref-like makes many cases, like array access or tuple literals are never applicable due to ref-like constraints.
+
+                    //PROTOTYPE: __refvalue. It is not possible to __makeref(span), but it is allowed to do __refvalue(tr, Span), perhaps it should not be allowed, looks like a bug.
+                    //           otherwise we need to decide how to treat that. Most likely as scopeOfTheContainingExpression
+            }
```

Comment:
This switch-case reminds me that we should update the compiler test plan with some notes for ref features. It seems likely that adding new bound nodes will require new logic here.

Reply:
It would affect only expressions though.

---
In reply to: [135363754](https://github.com/dotnet/roslyn/pull/21334#discussion_r135363754) [](ancestors = 135363754)

Reply:
Maybe add an assert to help catch new expressions?

Reply:
Eventually. Fornow we know that the switch is not comprehensive.

---
In reply to: [135366336](https://github.com/dotnet/roslyn/pull/21334#discussion_r135366336) [](ancestors = 135366336,135363754)

Reply:
I will add a //PROTOTYPE comment to consider that.

---
In reply to: [135666388](https://github.com/dotnet/roslyn/pull/21334#discussion_r135666388) [](ancestors = 135666388,135366336,135363754)